packages/fastify-siwe/src/registerSiweRoutes.ts

```diff
@@ -0,0 +1,78 @@
+import type { FastifyInstance, FastifyReply, FastifyRequest } from 'fastify'
+import { SessionStore } from './types'
+import { SiweMessage } from 'siwe'
+import { siweAuthenticated } from './index'
+
+export interface RegisterSiweRoutesOpts {
+  store: SessionStore
+}
+
+export const registerSiweRoutes = (fastify: FastifyInstance, { store }: RegisterSiweRoutesOpts) => {
+  fastify.post(
+    '/siwe/init',
+    {},
+    async function handler(this: FastifyInstance, req: FastifyRequest, reply: FastifyReply) {
+      reply.send({
+        nonce: await req.siwe.generateNonce(),
+      })
+    }
+  )
+
+  fastify.post(
+    '/siwe/cookie',
+    {},
+    async function handler(
+      this: FastifyInstance,
+      req: FastifyRequest<{
+        Body: {
+          signature: string
+          message: SiweMessage
+        }
+      }>,
+      reply: FastifyReply
+    ) {
+      const authToken = JSON.stringify({
+        message: req.body.message,
+        signature: req.body.signature,
+      })
+
+      reply
+        .setCookie('authToken', authToken, {
+          httpOnly: true,
+          secure: process.env.NODE_ENV === 'production',
```

Comment:
```suggestion
          secure: process.env.NODE_ENV !== 'development',
```

Better to edge on the side of security by default if `NODE_ENV` is not set.